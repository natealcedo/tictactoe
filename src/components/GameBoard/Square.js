import React from "react";
import propTypes from "prop-types";
import "./square.css";

const Square = ({ value, onClick, row, column }) => {
  if (value === null) {
    return <div className="square" onClick={() => onClick(row, column)} />;
  }

  if (value === 0) {
    return <div className="square">X</div>;
  }

  return <div className="square">O</div>;
};

Square.propTypes = {
  value: propTypes.oneOf([0, 1]),
  onClick: propTypes.func.isRequired,
  row: propTypes.number.isRequired,
  column: propTypes.number.isRequired
};

export default Square;
