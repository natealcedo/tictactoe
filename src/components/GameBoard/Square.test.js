import React from "react";
import Square from "./Square";
import { shallow } from "enzyme";

describe("Square", () => {
  each([0, 1, null]).test(
    "should render without exploding for value %p",
    value => {
      const wrapper = shallow(
        <Square row={0} column={0} onClick={() => {}} value={value} />
      );
      expect(wrapper).toMatchSnapshot();
    }
  );

  test("should call onclick when clicked", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Square row={0} column={0} onClick={onClick} value={null} />
    );
    wrapper.simulate("click");
    expect(onClick).toHaveBeenCalled();
  });
});
