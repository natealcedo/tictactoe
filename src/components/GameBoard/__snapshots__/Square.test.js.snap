// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should render without exploding for value 0 1`] = `
<div
  className="square"
>
  X
</div>
`;

exports[`Square should render without exploding for value 1 1`] = `
<div
  className="square"
>
  O
</div>
`;

exports[`Square should render without exploding for value null 1`] = `
<div
  className="square"
  onClick={[Function]}
/>
`;
